torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -221,6 +221,7 @@ class NCCLComm {
       int numRanks,
       int rank,
       std::vector<ncclUniqueId>& commIds,
+      at::DeviceIndex deviceIndex,
```

Comment:
I don't think this constructor was deliberately missing the device?

Reply:
lgtm. Yeah, the other "create" methods take `deviceIndex`.